docs/sql-ref-number-pattern.md

```diff
@@ -19,4 +19,101 @@ license: |
   limitations under the License.
 ---
 
-TODO: Add the content of Number Patterns for Formatting and Parsing
+### Description
+
+Functions such as `to_number` and `try_to_number` support converting between values of string and
+Decimal type. Such functions accept format strings indicating how to map between these types.
+
+### Syntax
```

Comment:
Shall we have the highlights for all the functions?
E.g.
```
to_number(string, format);
try_to_number(string, format);
```

Reply:
Sure, this sounds good. Added this to the docs.